quickstarts/deploy-vm/main.go

```diff
@@ -0,0 +1,159 @@
+package main
+
+import (
+	"context"
+	"encoding/json"
+	"io/ioutil"
+	"log"
+
+	"github.com/Azure/azure-sdk-for-go/profiles/2017-03-09/network/mgmt/network"
+	"github.com/Azure/azure-sdk-for-go/profiles/2017-03-09/resources/mgmt/resources"
+
+	"github.com/Azure/go-autorest/autorest"
+	"github.com/Azure/go-autorest/autorest/adal"
+	"github.com/Azure/go-autorest/autorest/azure"
+	"github.com/Azure/go-autorest/autorest/to"
+)
+
+type authInfo struct {
+	TenantID               string
+	SubscriptionID         string
+	ServicePrincipalID     string
+	ServicePrincipalSecret string
+}
+
+const (
+	resourceGroupName     = "GoVMQuickstart"
+	resourceGroupLocation = "eastus"
+
+	deploymentName = "VMDeployQuickstart"
+	templateFile   = "vm-quickstart-template.json"
+	parametersFile = "vm-quickstart-params.json"
+)
+
+var (
+	config = authInfo{
+		TenantID:               "",
+		SubscriptionID:         "",
+		ServicePrincipalID:     "",
+		ServicePrincipalSecret: "",
+	}
+
+	ctx = context.Background()
+
+	token *adal.ServicePrincipalToken
+)
+
+// Authenticate with the Azure services over OAuth, using a service principal.
+func init() {
+	oauthConfig, err := adal.NewOAuthConfig(azure.PublicCloud.ActiveDirectoryEndpoint, config.TenantID)
+	if err != nil {
+		log.Fatalf("%s: %v\n", "failed to get OAuth config", err)
+	}
+	token, err = adal.NewServicePrincipalToken(
+		*oauthConfig,
+		config.ServicePrincipalID,
+		config.ServicePrincipalSecret,
+		azure.PublicCloud.ResourceManagerEndpoint)
+	if err != nil {
+		log.Fatalf("faled to get token: %v\n", err)
+	}
+}
+
+func main() {
+	group, err := createGroup()
+	if err != nil {
+		log.Fatalf("failed to create group: %v", err)
+	}
+	log.Printf("created group: %v\n", *group.Name)
+
+	log.Printf("starting deployment\n")
+	result, err := createDeployment()
+	if err != nil {
+		log.Fatalf("Failed to deploy correctly: %v", err)
+	}
+	log.Printf("Completed deployment: %v", *result.Name)
+	getLogin()
+}
+
+// Create a resource group for the deployment.
+func createGroup() (resources.Group, error) {
+	groupsClient := resources.NewGroupsClient(config.SubscriptionID)
+	groupsClient.Authorizer = autorest.NewBearerAuthorizer(token)
+
+	return groupsClient.CreateOrUpdate(
+		ctx,
+		resourceGroupName,
+		resources.Group{
+			Location: to.StringPtr(resourceGroupLocation)})
+}
+
+// Create the deployment
+func createDeployment() (resources.DeploymentExtended, error) {
```

Comment:
>resources.DeploymentExtended, error [](start = 25, length = 35)

Return types can have names (func signature would look like this `func createDeployment() (de resources.DeploymentExtended, err error)`, so you can avoid `return resources.DeploymentExtended{}, err`, and just `return`

Reply:
Wouldn't this require creating a `de` variable with an empty value? I would rather have the explicit return.